File: trunk/src/ConstraintSolver/DOF.cpp
/****************************************************************************
**
** This file is part of the pSketcher project.
**
** This file may be used under the terms of the GNU General Public
** License version 2.0 as published by the Free Software Foundation
** and appearing in the file LICENSE.GPL included in the packaging of
** this file.
**
** This file is provided AS IS with NO WARRANTY OF ANY KIND, INCLUDING THE
** WARRANTY OF DESIGN, MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE.
**
** Copyright (C) 2006-2009 Michael Greminger. All rights reserved.
**
****************************************************************************/

#include <string>
#include <sstream>
#include <iostream>

#include "DOF.h"

using namespace std;

// Initialize private static variables for DOF and PrimitiveBase classes
unsigned DOF::next_id_number_ = 1;

DOF::DOF (bool free, bool dependent) :
id_number_(next_id_number_++),free_(free), dependent_(dependent),
database_(0),
delete_me_(false)
{
	// by default, name variable using id_number_
	stringstream variable_name;
	variable_name << "dof" << id_number_;

	name_ = variable_name.str();
}

DOF::DOF ( const char *name, bool free, bool dependent) :
id_number_(next_id_number_++),free_(free), dependent_(dependent),
database_(0),
delete_me_(false)
{
	name_ = name;
}

DOF::DOF (unsigned id, bool dependent) :
id_number_(id),free_(false), dependent_(dependent),
database_(0),
delete_me_(false)
{

}

File: trunk/src/mono/ConstraintSolver/DOF.cs
//
//  DOF.cs
//
//  Author:
//       grem <${AuthorEmail}>
//
//  Copyright (c) 2009 grem
//
//  This program is free software; you can redistribute it and/or modify
//  it under the terms of the GNU General Public License as published by
//  the Free Software Foundation; either version 2 of the License, or
//  (at your option) any later version.
//
//  This program is distributed in the hope that it will be useful,
//  but WITHOUT ANY WARRANTY; without even the implied warranty of
//  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
//  GNU General Public License for more details.
//
//  You should have received a copy of the GNU General Public License
//  along with this program; if not, write to the Free Software
//  Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA 02111-1307 USA
//

using System;

namespace pSketcher.ConstraintSolver
{
    /// <summary>
    /// Abstract base class for all DOF's
    /// </summary>
	abstract class DOF
	{
        //
        // Field Data
        //
        static uint next_id_number_ = 0;

		public DOF (bool free)
		{
            this.Id = checked(next_id_number_++);
            this.Free = free;
            this.Name = "dof_" + Id;
		}


        //
        // Properties and assecors
        //
        public abstract double Value {get; set;}
        public string Name {get; private set;}
        public bool Free {get; set;}
        public uint Id { get; private set; }

        //
        // Methods used by all instances
        //

	}
}
